subsys/bluetooth/host/gatt.c

```diff
@@ -52,6 +52,17 @@
 
 static uint16_t last_static_handle;
 
+#if defined(CONFIG_BT_SETTINGS)
```

Comment:
Either remove the #ifdef, or (perhaps better) move this definition down to where it's actually used (and probably already behind the same CONFIG_BT_SETTINGS #ifdef.